server/apps/microservices/src/processors/metadata-extraction.processor.ts

```diff
@@ -91,7 +92,8 @@ export class MetadataExtractionProcessor {
     let asset = job.data.asset;
 
     try {
-      const exifData = await exiftool.read<ImmichTags>(asset.originalPath).catch((error: any) => {
+      const sidecarPath = asset.sidecarPath || (fs.existsSync(`${asset.originalPath}.xmp`) ? `${asset.originalPath}.xmp` : null)
```

Comment:
This works, but I'd prefer using the async method here.

Reply:
We use this elsewhere now, but this was resolved using the async `fs.promises.access` method. Using access instead of exists as exists is now deprecated in the node docs and recommends using access if checking file access without making modifications.